src/main/java/hudson/security/GlobalMatrixAuthorizationStrategy.java

```diff
@@ -473,12 +473,16 @@ public FormValidation doCheckName_(@Nonnull String value, @Nonnull AccessControl
 
             if(v.equals("authenticated"))
                 // system reserved group
-                return FormValidation.respond(Kind.OK, makeImg("user.png", "Group", false) +ev);
+                return FormValidation.respond(Kind.OK, makeImg("user.png", ev, "Group", false));
 
             try {
                 try {
                     sr.loadUserByUsername(v);
-                    return FormValidation.respond(Kind.OK, makeImg("person.png", "User", false)+ev);
+                    User u = User.get(v);
+                    if (ev.equals(u.getFullName())) {
+                        return FormValidation.respond(Kind.OK, makeImg("person.png", ev, "User", false));
+                    }
+                    return FormValidation.respond(Kind.OK, makeImg("person.png", u.getFullName(), "User " + ev, false));
```

Comment:
This name is unlimited. I can imagine cases where the REALLY long name breaks the UI. The worst thing I have seem on my instance (not the one you would think about, another one) was somewhat more than 64 symbols because it was corrupting our user database in SCM